include/ck/utility/data_type.hpp

```diff
@@ -942,125 +942,160 @@ using int8x16_t = typename vector_type<int8_t, 16>::type;
 using int8x32_t = typename vector_type<int8_t, 32>::type;
 using int8x64_t = typename vector_type<int8_t, 64>::type;
 
-// Convert X to Y
-template <typename Y, typename X>
-__host__ __device__ constexpr Y type_convert(X x)
+class TypeConvert
 {
-    static_assert(!std::is_reference_v<Y> && !std::is_reference_v<X>);
+    public:
+    // constructor
+    __host__ __device__ TypeConvert()
+    {
+        BF16ConvertRTN_ = false; // use round to zero by default
+    }
 
-    return static_cast<Y>(x);
-}
+    // switch bf16 conversion mode to rtn
+    __host__ __device__ void SetBF16ConvertRTN() { BF16ConvertRTN_ = true; }
 
-// convert bfp16 to fp32
-template <>
-inline __host__ __device__ constexpr float type_convert<float, bhalf_t>(bhalf_t x)
-{
-    union
+    // switch bf16 conversion mode to rtz
+    __host__ __device__ void SetBF16ConvertRTZ() { BF16ConvertRTN_ = false; }
+
+    // convert for all types except bf16
+    template <typename Y, typename X>
+    __host__ __device__ constexpr Y convert(X x)
```

Comment:
We can use 'operator()' overloading here.
```
    __host__ __device__ Y operator()(X& x) const
    {
        return static_cast<Y>(x);
    }
```

Reply:
Oh, nice call, thanks!